File: app/src/main/java/com/weiqi/ui/components/ZenComponents.kt
package com.weiqi.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/** Paper-tone card. Soft 14dp rounding for the "carved wood" feel. */
@Composable
fun ZenCard(
    modifier: Modifier = Modifier,
    container: Color = MaterialTheme.colorScheme.surfaceContainer,
    contentPadding: PaddingValues = PaddingValues(16.dp),
    content: @Composable () -> Unit
) {
    Box(
        modifier = modifier
            .clip(RoundedCornerShape(14.dp))
            .background(container)
            .padding(contentPadding)
    ) { content() }
}

/**
 * Pill-shaped tag. Picks a contrasting text color automatically based on the container.
 */
@Composable
fun ZenChip(
    text: String,
    modifier: Modifier = Modifier,
    container: Color? = null,
    contentColor: Color? = null
) {
    val bg = container ?: MaterialTheme.colorScheme.secondaryContainer
    val fg = contentColor ?: contentColorFor(bg)
    Box(
        modifier = modifier
            .clip(RoundedCornerShape(999.dp))
            .background(bg)
            .padding(horizontal = 12.dp, vertical = 6.dp)
    ) {
        Text(
            text = text,
            style = MaterialTheme.typography.labelSmall,
            color = fg
        )
    }
}

private fun contentColorFor(bg: Color): Color {
    // Simple luminance check — dark backgrounds get light text.
    val l = 0.299 * bg.red + 0.587 * bg.green + 0.114 * bg.blue
    return if (l < 0.5) Color(0xFFFFF8F3) else Color(0xFF221A10)
}

File: lib/src/ui/components/zen_components.dart
import 'package:flutter/material.dart';

import '../theme.dart';

/// Paper-tone card with a restrained shared corner radius.
class ZenCard extends StatelessWidget {
  final Widget child;
  final Color? container;
  final EdgeInsetsGeometry contentPadding;
  final VoidCallback? onTap;

  const ZenCard({
    super.key,
    required this.child,
    this.container,
    this.contentPadding = const EdgeInsets.all(16),
    this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final bg = container ?? scheme.surfaceContainer;
    final card = Material(
      color: bg,
      borderRadius: BorderRadius.circular(Zen.cardRadius),
      clipBehavior: Clip.antiAlias,
      child: onTap == null
          ? Padding(padding: contentPadding, child: child)
          : InkWell(
              onTap: onTap,
              child: Padding(padding: contentPadding, child: child),
            ),
    );
    return card;
  }
}

/// Pill-shaped tag. Picks a contrasting text color automatically based on the container.
class ZenChip extends StatelessWidget {
  final String text;
  final Color? container;
  final Color? contentColor;

  const ZenChip({
    super.key,
    required this.text,
    this.container,
    this.contentColor,
  });

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final bg = container ?? scheme.secondaryContainer;
    final fg = contentColor ?? _contentColorFor(bg);
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
      decoration: BoxDecoration(
        color: bg,
        borderRadius: BorderRadius.circular(999),
      ),
      child: Text(
        text,
        style: Theme.of(context).textTheme.labelSmall?.copyWith(color: fg),
      ),
    );
  }

  static Color _contentColorFor(Color bg) {
    final l = 0.299 * bg.r + 0.587 * bg.g + 0.114 * bg.b;
    return l < 0.5 ? const Color(0xFFFFF8F3) : const Color(0xFF221A10);
  }
}

class ZenOptionButton extends StatelessWidget {
  final String label;
  final bool selected;
  final VoidCallback onTap;

  const ZenOptionButton({
    super.key,
    required this.label,
    required this.selected,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final bg = selected ? scheme.primary : scheme.surfaceContainerHigh;
    final fg = selected ? scheme.onPrimary : scheme.onSurface;
    return Material(
      color: bg,
      borderRadius: BorderRadius.circular(Zen.controlRadius),
      clipBehavior: Clip.antiAlias,
      child: InkWell(
        onTap: onTap,
        child: SizedBox(
          height: 44,
          child: Center(
            child: Text(
              label,
              overflow: TextOverflow.ellipsis,
              maxLines: 1,
              style: Theme.of(context)
                  .textTheme
                  .labelMedium
                  ?.copyWith(color: fg),
            ),
          ),
        ),
      ),
    );
  }
}
